openfga/v1/openfga.proto

```diff
@@ -32,6 +32,57 @@ message Object {
   ];
 }
 
+// User represents any possible value for a user (subject)
+//
+// Can either be a:
+// - Specific user object (ex: 'user:will')
+// - Specific userset (ex: 'group:engineering#member')
+// - Public-typed wildcard (ex: 'user:*')
+//
+// See https://docs.fga.dev/intro/okta-fga-concepts#what-is-a-user
+message User {
+  oneof user {
+    Object object = 1;
+    UsersetUser userset = 2;
+    TypedWildcard wildcard = 3;
+  }
+}
+
+message ObjectOrUserset {
+  oneof user {
+    Object object = 1;
+    UsersetUser userset = 2;
+  }
+}
+
+message UsersetUser {
+  string type = 1 [
+    (validate.rules).string = {
+      pattern: "^[^:#@\\s]{1,254}$",
+      ignore_empty: false
+    },
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"group\""}
+  ];
+
+  string id = 2 [
+    (validate.rules).string = {
+      pattern: "[^#:\\s]+$",
+      ignore_empty: false
+    },
+    (google.api.field_behavior) = REQUIRED
+  ];
```

Comment:
Nitpick, non-blocking: add examples for these too.

<img width="485" alt="image" src="https://github.com/openfga/api/assets/5374887/7282f97a-b279-4b82-9516-ddfbf2894265">

Reply:
Will address in follow-up.